source/common/res/features/running-balance/main.js

```diff
@@ -0,0 +1,162 @@
+              continue;
+            }
+
+            if (transaction.get('inflow')) {
+              runningBalance += transaction.get('inflow');
+            } else if (transaction.get('outflow')) {
+              runningBalance -= transaction.get('outflow');
+            }
+
+            transaction.__ynabToolKitRunningBalance = runningBalance;
+          }
+        }
+      }
+
+      function updateRunningBalanceColumn() {
+        insertHeader();
+
+        $('.ynab-grid-container .ynab-grid-body .ynab-grid-body-row').each(function (index, element) {
+          var $element = $(element);
+          if ($element.hasClass('ynab-grid-body-empty')) return;
+
+          insertValue(element);
+        });
+      }
+
+      function insertHeader() {
+        if ($('.ynab-grid-header .ynab-toolkit-grid-cell-running-balance').length) return;
+
+        var $headerRow = $('.ynab-grid-header');
+        var runningBalanceHeader = $('.ynab-grid-cell-inflow', $headerRow).clone();
+        runningBalanceHeader.removeClass('ynab-grid-cell-inflow');
+        runningBalanceHeader.addClass('ynab-toolkit-grid-cell-running-balance');
+        runningBalanceHeader.text('RUNNING BALANCE');
+        runningBalanceHeader.insertAfter($('.ynab-grid-cell-inflow', $headerRow));
+
+        if ($('.ynab-grid-body .ynab-grid-body-row-top .ynab-toolkit-grid-cell-running-balance').length) return;
+        var $topRow = $('.ynab-grid-body-row-top');
+        var topRowRunningBalance = $('.ynab-grid-cell-inflow', $topRow).clone();
+        topRowRunningBalance.removeClass('ynab-grid-cell-inflow');
+        topRowRunningBalance.addClass('ynab-toolkit-grid-cell-running-balance');
+        topRowRunningBalance.insertAfter($('.ynab-grid-cell-inflow', $topRow));
+      }
+
+      function insertValue(element) {
+        var $currentRow = $(element);
+        var currentRowRunningBalance = $('.ynab-grid-cell-inflow', $currentRow).clone();
+        currentRowRunningBalance.removeClass('ynab-grid-cell-inflow');
+        currentRowRunningBalance.addClass('ynab-toolkit-grid-cell-running-balance');
+
+        var emberView = ynabToolKit.shared.getEmberView($currentRow.attr('id'));
+        var transaction = emberView.get('content');
+        var runningBalance = transaction.get('__ynabToolKitRunningBalance');
+        var currencySpan = $('.user-data', currentRowRunningBalance);
+        var currencySymbol = ynab.YNABSharedLib.currencyFormatter.getCurrency().currency_symbol;
+
+        if (runningBalance < 0) {
+          currencySpan.addClass('user-data currency negative');
+        } else if (runningBalance > 0) {
+          currencySpan.addClass('user-data currency positive');
+        } else {
+          currencySpan.addClass('user-data currency zero');
+        }
+
+        if (transaction.get('parentEntityId') !== null) {
+          currencySpan.text('');
+        } else {
+          currencySpan.text(currencySymbol + ynab.formatCurrency(runningBalance));
```

Comment:
Hate to be "that guy" but sometimes currencies have the symbol after the number. For example 3.56£. We have a shared library function called `formatCurrency` [here](https://github.com/toolkit-for-ynab/toolkit-for-ynab/blob/master/source/common/res/features/shared/main.js#L137) that should help. The source for that function came from the minified version of YNAB's own code to do this.